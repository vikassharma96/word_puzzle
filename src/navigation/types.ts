import type {NativeStackScreenProps} from '@react-navigation/native-stack';

export type RootStackParamList = {
  Home: undefined;
  Puzzle: {categoryId: string};
  Score: undefined;
  LeadersBoard: undefined;
};

export type HomeScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'Home'
>;
export type PuzzleScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'Puzzle'
>;
export type ScoreScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'Score'
>;
export type LeadersBoardScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'LeadersBoard'
>;
