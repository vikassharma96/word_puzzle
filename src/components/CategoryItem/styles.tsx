import {StyleSheet} from 'react-native';
import colors from '../../styles/colors';
import defaultStyles from '../../styles/defaultStyles';

const styles = StyleSheet.create({
  container: {
    marginVertical: '3%',
    alignItems: 'center',
    alignSelf: 'center',
    minWidth: '50%',
    ...defaultStyles.button,
  },
  selected: {
    backgroundColor: colors.light.button,
  },
});

export default styles;
