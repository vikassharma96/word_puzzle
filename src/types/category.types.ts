type ICategory = {
  id: string;
  name: string;
};

export type {ICategory};
