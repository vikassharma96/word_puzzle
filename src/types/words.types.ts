type IWords = {
  word: string;
  description: string;
};

export type {IWords};
