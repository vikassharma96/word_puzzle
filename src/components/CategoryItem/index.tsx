import React from 'react';
import Button from '../Button';
import styles from './styles';
import colors from '../../styles/colors';
import {useThemeColor} from '../../hooks/useTheme';
import {ICategory} from '../../types/category.types';

type CategoryItemProps = {
  category: ICategory;
  selectedCategoryId?: string;
  setSelectedCategoryId: (categoryId: string) => void;
};

const CategoryItem = ({
  category,
  selectedCategoryId,
  setSelectedCategoryId,
}: CategoryItemProps) => {
  const {id: categoryId, name: categoryName} = category;
  const selectedBackgroundColor = useThemeColor('button');

  const handleOnCategory = () => {
    setSelectedCategoryId(categoryId);
  };

  return (
    <Button
      title={categoryName}
      style={styles.container}
      testID={'item'}
      onPress={handleOnCategory}
      backgroundColor={
        selectedCategoryId === categoryId
          ? selectedBackgroundColor
          : colors.default.background
      }
    />
  );
};

export default CategoryItem;
