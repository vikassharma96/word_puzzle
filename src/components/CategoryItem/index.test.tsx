import React from 'react';
import CategoryItem from '.';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';
import {ICategory} from '../../types/category.types';

describe('@components/CategoryItem', () => {
  const cateogry: ICategory = {
    id: '123',
    name: 'Food',
  };
  const selectedCatIdMock = jest.fn();
  const renderComponent = (selectedCategoryId?: string) => {
    return (
      <CategoryItem
        category={cateogry}
        selectedCategoryId={selectedCategoryId}
        setSelectedCategoryId={selectedCatIdMock}
      />
    );
  };

  it('should match snapshots', () => {
    const componentTree = renderer.create(renderComponent()).toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  it('should be able to change background if selected', () => {
    const {getByTestId} = render(renderComponent('123'));
    const element = getByTestId('item');
    expect(element).toBeDefined();
  });

  it('should be able to click on the item', () => {
    const {getByTestId} = render(renderComponent());
    const element = getByTestId('item');
    expect(element).toBeDefined();
    fireEvent.press(element);
    expect(selectedCatIdMock).toHaveBeenCalled();
  });
});
