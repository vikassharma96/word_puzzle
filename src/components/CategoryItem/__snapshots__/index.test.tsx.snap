// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/CategoryItem should match snapshots 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#fff",
      "borderRadius": 6,
      "borderWidth": 1.5,
      "marginHorizontal": 8,
      "marginVertical": "3%",
      "minWidth": "50%",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 6,
      "textAlign": "center",
    }
  }
  testID="item"
>
  <Text
    style={
      [
        {
          "fontFamily": "Poppins-Medium",
          "fontSize": 14,
        },
        {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    Food
  </Text>
</View>
`;
